src/librustc_mir/interpret/terminator.rs

```diff
@@ -245,7 +245,9 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         trace!("eval_fn_call: {:#?}", fn_val);
 
         let instance = match fn_val {
-            FnVal::Instance(instance) => instance,
+            FnVal::Instance(instance) => {
+                self.subst_and_normalize_erasing_regions_in_frame(instance)?
```

Comment:
The source of this `instance` could be either a function pointer or something taken from the MIR. Is this substitution really right when it is a function pointer? I would think not, because the function pointer should already be monomorphized -- and it has nothing to do with the current frame.

So, just based on that intuition, the substitution should really happen [here](https://github.com/rust-lang/rust/blob/754c14c858c57388644565cfe2b4e95df4168d81/src/librustc_mir/interpret/terminator.rs#L88) instead.